### PR TITLE
Bot promotion desktop 947

### PR DESCRIPTION
**Pull request by MarcoPolo**
@gabriel 

(haven't tested it yet, just want a review before. Will test before merging)

Related PR on client: https://github.com/keybase/client/pull/2743

### CODE REVIEW

File: launchd/promotearelease.sh [hunk 1]
```diff
@@ -21,6 +21,15 @@ err_report() {
...
+if [ -n "$RELEASE_TO_PROMOTE" ];
+then
+  "$release_bin" promote-a-release --release="$RELEASE_TO_PROMOTE" --bucket-name="$bucket_name" --platform="$platform"
+  "$client_dir/packaging/slack/send.sh" "Promoted $platform release $RELEASE_TO_PROMOTE ($bucket_name)"
+else
+  "$release_bin" promote-releases --bucket-name="$bucket_name" --platform="$platform"
+  "$client_dir/packaging/slack/send.sh" "Promoted $platform release on ($bucket_name)"
+fi
+
+# TODO (mm / gh) this should happen automatically when we promote a release, otherwise it's easy to forget.
```
Comment (MarcoPolo) [inline launchd/promotearelease.sh]: @gabriel ^ any reason why this doesn't happen automatically?

File: keybot/main.go [hunk 2]
```diff
@@ -91,7 +91,7 @@ func kingpinKeybotHandler(args []string) (string, error) {
 		if err != nil {
 			return "", err
 		}
-		return slackbot.NewExecCommand("/bin/launchctl", []string{"start", "keybase.prerelease.promotearelease"}, false, "Promote a specific release to public").Run(emptyArgs)
+		return slackbot.NewExecCommand("/bin/launchctl", []string{"start", "keybase.prerelease.promotearelease"}, false, "Promote a release to public, takes an optional specific release").Run(emptyArgs)
```
Comment (gabriel) [inline keybot/main.go]: is release optional?

File: launchd/keybase.keybot.promote.darwin.plist [hunk 3]
```diff
@@ -0,0 +1,62 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+    <key>Label</key>
+    <string>keybase.keybot.promote.darwin</string>
```
Comment (gabriel) [inline launchd/keybase.keybot.promote.darwin.plist]: i don't think you need darwin here cause launchd plists are only darwin?

When changing here be sure to update log files.

File: launchd/promotearelease.sh [hunk 4]
```diff
@@ -21,6 +21,15 @@ err_report() {
 
 trap 'err_report $LINENO' ERR
 
-"$release_bin" promote-a-release --release="$RELEASE_TO_PROMOTE" --bucket-name="$bucket_name" --platform="$platform"
 
-"$client_dir/packaging/slack/send.sh" "Promoted $platform release $RELEASE_TO_PROMOTE ($bucket_name)"
+if [ -n "$RELEASE_TO_PROMOTE" ];
```
Comment (gabriel) [inline launchd/promotearelease.sh]: Could do `if [ -n "$RELEASE_TO_PROMOTE" ]; then`

File: launchd/promotearelease.sh (hunk 1 above)
Comment (gabriel) [inline launchd/promotearelease.sh]: yep kill and have it done at end of all promotes

**Comment by gabriel** [comment]
lg2m with comments

File: keybot/main.go (hunk 2 above)
Comment (MarcoPolo) [inline keybot/main.go]: yes, if a release isn't passed in then it just does the `promote-releases` command.

File: launchd/keybase.keybot.promote.darwin.plist (hunk 3 above)
Comment (MarcoPolo) [inline launchd/keybase.keybot.promote.darwin.plist]: I was following the convention of keybase.keybot.[thing].[platform].plist. like `keybase.keybot.build.darwin`. It's possible to have keybot sending a promote command for other platforms too.

Comment (gabriel) [inline launchd/keybase.keybot.promote.darwin.plist]: yep fine

**Pull request merged**